In [1]:
#All imports

import pandas as pd
from google.colab import files
uploaded = files.upload()

Saving problem_set_1_data.csv to problem_set_1_data (2).csv


###Block 2: Load data, rename columns, print head, tail and describe

In [2]:
data = pd.read_csv('problem_set_1_data.csv')

In [3]:
data.columns=['quiz_total', 'attended', 'final_grade', 'final_exam','name'] 
print(data.head)
print(data.tail)
print(data.describe())

<bound method NDFrame.head of      quiz_total  attended  final_grade  final_exam                 name
0         78.50         1        99.38          46         Linnie Lietz
1         64.25         0        95.78          45        Mitch Mustain
2         76.75         0        97.42          41       Salina Chavera
3         69.00         1        92.50          38    Kimberely Conwell
4         64.00         0        92.50          39            Zack Burk
..          ...       ...          ...         ...                  ...
166       73.50         0        96.25          45           Dana Malta
167       75.50         0        98.13          44         Sixta Heyden
168       41.00         0        69.38          32       Brianne Broome
169       77.00         0       101.25          47  Elouise Weatherholt
170       72.00         0        92.81          38        Addie Maharaj

[171 rows x 5 columns]>
<bound method NDFrame.tail of      quiz_total  attended  final_grade  final_exam 

The mean describes the average score of each category in the course of every student. For quiz_total, the average score was 69.13 out of the maximum score which is 79.5. The average participation point for attending class was 0.19. The average final grade was a 91.85  with the highest student having 104.38. The average grade for the final exam is 39.67, with the highest scorer having 49 points. 

###Block 3: Subset data by whether someone attended class

In [4]:
#creates a new dataframe from data for those who attends class
attended = data[data["attended"] == 1]

#creates a new dataframe from data for those who dont attend class
absent = data[data["attended"] == 0]

In [5]:
attended.describe()

quiz_total  attended  final_grade  final_exam
count   33.000000      33.0    33.000000   33.000000
mean    72.500000       1.0    95.072121   40.636364
std      5.547029       0.0     4.673980    4.014179
min     54.000000       1.0    85.470000   33.000000
25%     70.000000       1.0    91.250000   38.000000
50%     74.000000       1.0    94.690000   40.000000
75%     77.000000       1.0    99.380000   44.000000
max     78.500000       1.0   101.250000   47.000000

In [6]:
absent.describe()

quiz_total  attended  final_grade  final_exam
count  138.000000     138.0   138.000000  138.000000
mean    68.326087       0.0    91.086159   39.442029
std     10.057028       0.0    10.174000    5.800853
min      0.000000       0.0     2.500000    0.000000
25%     65.000000       0.0    88.320000   37.000000
50%     71.000000       0.0    92.770000   40.000000
75%     75.375000       0.0    96.210000   43.750000
max     79.500000       0.0   104.380000   49.000000

For those that attended class, the average quiz_total is: 72.5, the average final_grade is: 95, the average final_exam is 40.6

For those that were absent/didn't attend class, the average quiz_total was: 68.33, the average_final grade is 91, the average final_exam is: 39.44.

### Block 4: Extra credit

In [7]:
working_tgt = data[data.duplicated(['quiz_total','final_grade','final_exam'],
                                   keep= False)]
print(working_tgt)

     quiz_total  attended  final_grade  final_exam           name
146        72.0         0        92.81          38  Jaraham Eidda
170        72.0         0        92.81          38  Addie Maharaj
